### PR TITLE
created example for route tta functions

### PR DESCRIPTION
**Pull request by silvestrpredko**

### CODE REVIEW

File: route-tta/app/src/main/java/com/here/android/example/route/tta/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+import com.here.android.mpa.routing.RoutingError;
+
+import java.io.File;
+import java.util.List;
+
+public class MainActivity extends AppCompatActivity {
+
+    private final int REQUEST_CODE = 0x754;
+
+    private MapView mapView;
```
Comment (NazarKacharaba) [inline route-tta/app/src/main/java/com/here/android/example/route/tta/MainActivity.java]: Try using code style that is used across all here android samples. E.g. class variables should have `m_` prefix, etc.

File: route-tta/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+android {
+    compileSdkVersion 28
+    defaultConfig {
+        applicationId "com.here.android.example.route.tta"
+        minSdkVersion 16
+        targetSdkVersion 28
+        versionCode 1
+        versionName "1.0"
+        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
```
Comment (starand) [inline route-tta/app/build.gradle]: no need for this

File: route-tta/app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,31 @@
+apply plugin: 'com.android.application'
+
+android {
+    compileSdkVersion 28
```
Comment (starand) [inline route-tta/app/build.gradle]: add empty lines between sections.

File: route-tta/app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        <service
+            android:name="com.here.android.mpa.service.MapService"
+            android:label="{YOUR_LABEL_NAME}"
+            android:exported="false">
+            <intent-filter>
+                <action android:name="{YOUR_INTENT_NAME}"></action>
+            </intent-filter>
+        </service>
+    </application>
+</manifest>
```
Comment (starand) [inline route-tta/app/src/main/AndroidManifest.xml]: add new line

File: route-tta/app/src/main/java/com/here/android/example/route/tta/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+        setContentView(R.layout.activity_main);
+        m_mapView = findViewById(R.id.mapView);
+        m_handlerThread.start();
+        m_backgroundHandler = new Handler(m_handlerThread.getLooper());
+        setUp();
+    }
+
+    private void setUp() {
+        final int result = ContextCompat.checkSelfPermission(
+                this, Manifest.permission.WRITE_EXTERNAL_STORAGE
```
Comment (starand) [inline route-tta/app/src/main/java/com/here/android/example/route/tta/MainActivity.java]: add ');' on the same line

File: route-tta/app/src/main/java/com/here/android/example/route/tta/MainActivity.java [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+        );
+
+        // request permission if needed for external storage
+        if (result != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(
```
Comment (starand) [inline route-tta/app/src/main/java/com/here/android/example/route/tta/MainActivity.java]: reformat into two lines

File: route-tta/app/src/main/java/com/here/android/example/route/tta/MainActivity.java [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+
+    private void calculateTta() {
+        m_backgroundHandler.post(new Runnable() {
+            @Override
+            public void run() {
+                /*
+                 * Here received arrival time for the whole m_route, if you want to get time only
+                 * for part of m_route pass parameter in bounds 0 <= m_route.getSublegCount()
+                 */
+                final RouteTta tta_excluding = m_route.getTtaExcludingTraffic(Route.WHOLE_ROUTE);
```
Comment (starand) [inline route-tta/app/src/main/java/com/here/android/example/route/tta/MainActivity.java]: ttaExcluding - use camel case

File: route-tta/app/src/main/java/com/here/android/example/route/tta/MainActivity.java [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+    }
+
+    private void calculateTtaUsingDownloadedTraffic() {
+        // turn on traffic updates
+        TrafficUpdater.getInstance().enableUpdate(true);
+        m_requestInfo = TrafficUpdater.getInstance().request(
+                m_route, new TrafficUpdater.Listener() {
+                    @Override
+                    public void onStatusChanged(TrafficUpdater.RequestState requestState) {
+                        final RouteTta tta_using_downloaded = m_route.getTtaUsingDownloadedTraffic(
```
Comment (starand) [inline route-tta/app/src/main/java/com/here/android/example/route/tta/MainActivity.java]: use camel case

File: route-tta/app/src/main/java/com/here/android/example/route/tta/MainActivity.java [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+                                Route.WHOLE_ROUTE
+                        );
+
+                        if (callback != null) {
+                            runOnUiThread(new Runnable() {
+                                @Override
+                                public void run() {
+                                    callback.onTtaCalculated(
+                                            tta_using_downloaded.getDuration(),
```
Comment (starand) [inline route-tta/app/src/main/java/com/here/android/example/route/tta/MainActivity.java]: use camel case

File: route-tta/app/src/main/java/com/here/android/example/route/tta/MainActivity.java [hunk 10]
```diff
@@ -0,0 +1,248 @@
...
+                                            RouteUtil.TtaType.DOWNLOADED
+                                    );
```
Comment (starand) [inline route-tta/app/src/main/java/com/here/android/example/route/tta/MainActivity.java]: please reformat

File: route-tta/app/src/main/java/com/here/android/example/route/tta/MainActivity.java [hunk 11]
```diff
@@ -0,0 +1,248 @@
...
+                        }
+                    }
+                });
+    }
+
+    private RouteUtil.TtaCallback callback = new RouteUtil.TtaCallback() {
+        @Override
+        public void onTtaCalculated(final int time, RouteUtil.TtaType tta_type) {
+            TextView view;
+            switch (tta_type) {
```
Comment (starand) [inline route-tta/app/src/main/java/com/here/android/example/route/tta/MainActivity.java]: camel case

File: route-tta/app/src/main/java/com/here/android/example/route/tta/RouteUtil.java [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+        routeOptions.setTransportMode(RouteOptions.TransportMode.CAR);
+        /* Disable highway in this route. */
+        routeOptions.setHighwaysAllowed(false);
+        /* Calculate the shortest route available. */
+        routeOptions.setRouteType(RouteOptions.Type.FASTEST);
+        /* Calculate 1 route. */
+        routeOptions.setRouteCount(1);
+        /* Finally set the route option */
+        routePlan.setRouteOptions(routeOptions);
+
```
Comment (starand) [inline route-tta/app/src/main/java/com/here/android/example/route/tta/RouteUtil.java]: excessive lines

File: route-tta/app/src/main/java/com/here/android/example/route/tta/RouteUtil.java [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+
+
+        /* Define waypoints for the route */
+        /* START: Holländerstraße, Wedding, 13407 Berlin */
+        RouteWaypoint startPoint = new RouteWaypoint(
```
Comment (starand) [inline route-tta/app/src/main/java/com/here/android/example/route/tta/RouteUtil.java]: reformat the following tree method - remove excessive lines

File: route-tta/app/src/main/java/com/here/android/example/route/tta/RouteUtil.java [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+
+    static abstract class RouteListener<T, U extends Enum<?>> implements Router.Listener<T, U> {
+        @Override
+        public void onProgress(int i) {
+            /* The calculation progress can be retrieved in this callback. */
+        }
+    }
+
+    public enum TtaType {
+        INCLUDED, EXCLUDED, DOWNLOADED
```
Comment (starand) [inline route-tta/app/src/main/java/com/here/android/example/route/tta/RouteUtil.java]: write each entry from new line

File: route-tta/build.gradle [hunk 15]
```diff
@@ -0,0 +1,27 @@
+// Top-level build file where you can add configuration options common to all sub-projects/modules.
+
+buildscript {
+
```
Comment (starand) [inline route-tta/build.gradle]: excessive empty line

File: route-tta/build.gradle [hunk 16]
```diff
@@ -0,0 +1,27 @@
+// Top-level build file where you can add configuration options common to all sub-projects/modules.
+
+buildscript {
+    
+    repositories {
+        google()
+        jcenter()
+    }
+    dependencies {
```
Comment (starand) [inline route-tta/build.gradle]: add empty line

File: route-tta/build.gradle [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+        classpath 'com.android.tools.build:gradle:3.2.0'
+
```
Comment (starand) [inline route-tta/build.gradle]: excessive empty lines

File: route-tta/gradle.properties [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+# For more details on how to configure your build environment visit
+# http://www.gradle.org/docs/current/userguide/build_environment.html
+# Specifies the JVM arguments used for the daemon process.
+# The setting is particularly useful for tweaking memory settings.
+org.gradle.jvmargs=-Xmx1536m
+# When configured, Gradle will run in incubating parallel mode.
+# This option should only be used with decoupled projects. More details, visit
+# http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
+# org.gradle.parallel=true
+
```
Comment (starand) [inline route-tta/gradle.properties]: excessive empty line

**Pull request merged**